addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -305,8 +308,11 @@ export class Link extends Component {
         let isDocument = false;
         let directDownload = true;
         if (urlWithoutDomain && urlWithoutDomain.startsWith("/web/content/")) {
-            isDocument = true;
+            isDocument = !this.isLastAttachmentUrl;
             directDownload = urlWithoutDomain.includes("&download=true");
+            if (directDownload && !isDocument) {
+                this.state.url = this.state.url.replace("&download=true", "");
+            }
```

Comment:
If I understood correctly, the goal of this part is to remove `&download=true` from the url that is introduced by the `createElements()` method of the `DocumentSelector`. If it is so, what do you think of putting this part in the `save` part of the `openDocumentDialog`? I feel it would be better than relying on the call to `_getData()` to perform this operation :)